scene/main/node.cpp

```diff
@@ -154,14 +154,14 @@ void Node::_notification(int p_notification) {
 			data.in_constructor = false;
 		} break;
 		case NOTIFICATION_PREDELETE: {
-			set_owner(nullptr);
-
-			while (data.owned.size()) {
-				data.owned.front()->get()->set_owner(nullptr);
-			}
-
 			if (data.parent) {
-				data.parent->remove_child(this);
+				data.parent->remove_child(this); // This also removes the owner.
```

Comment:
Are you sure about this? From a quick look at `remove_child()` and the follow-up call to `_propagate_validate_owner()`, it doesn't seem strictly equivalent to the while loop above removing all owners.

Reply:
The node loses its parent, `_propagate_validate_owner` will search for current owner in node's ancestors and will remove the owner if it's not found (which will happen, because the parent was removed).

Although this probably needs to be tested with instanced scenes, because they have different owner...

Reply:
Ok, instanced scenes are broken. Here's a test case (replace code in OP):
```
$Node2D/Node.connect("tree_exiting", Callable(self, StringName("exiting")), [$Node2D/Node])
$Node2D/Node.connect("tree_exited", Callable(self, StringName("exited")), [$Node2D/Node])
$Node2D.free()
```
Node2D is an instanced scene with a Node child. Seems like Node will never lose its owner (prints the node in both cases, while previously it was both nulls).

Not sure how to fix this other than by forcing the removal of owner inside `remove_child` (which would require some hacks probably).